provider/kubernetes/kubernetes.go

```diff
@@ -188,6 +188,103 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 		}
 		templateObjects.TLS = append(templateObjects.TLS, tlsSection...)
 
+		if i.Spec.Backend != nil {
+			// If a global backend is configured, add this as a "default route" by adding a wildcard frontend, with the priority of the ingress.
+			defaultFrontendName := "global-default-frontend"
+			defaultBackendName := "global-default-backend"
+			if _, exists := templateObjects.Frontends[defaultFrontendName]; !exists {
+				// Ensure that we are not duplicating the frontend
+				if _, exists := templateObjects.Backends[defaultBackendName]; !exists {
+					// Ensure we are not duplicating the backend
+					templateObjects.Backends[defaultBackendName] = &types.Backend{
+						Servers: make(map[string]types.Server),
+						LoadBalancer: &types.LoadBalancer{
+							Method: "wrr",
+						},
+					}
+
+					service, exists, err := k8sClient.GetService(i.Namespace, i.Spec.Backend.ServiceName)
+					if err != nil {
+						log.Errorf("Error while retrieving service information from k8s API %s/%s: %v", i.Namespace, i.Spec.Backend.ServiceName, err)
```

Comment:
Because the error seems to be logged in a higher level function, WDYT to replace the `log.Errorf` by a `fmt.Errorf` and return this new error without logging it?

Reply:
Yes. On second thought, that makes sense